Sources/PackageLoading/PackageBuilder.swift

```diff
@@ -601,7 +601,7 @@ public struct PackageBuilder {
         
         // Create the test modules
         return try testsDirContents.filter(shouldConsiderDirectory).flatMap { dir in
-            return [try createModule(dir, name: dir.basename, isTest: true)]
+            return [try createModule(dir, name: dir.basename, isTest: true)].flatMap { $0 }
```

Comment:
Yes it is needed; without the flatMap it would try to do this: ``[[Module?]].flatMap -> [Module?]`` -- but the result should be ``[Module]`` instead. ``[[Module]?.flatMap -> [Module]].flatMap -> [Module]``.